### ISSUE TITLE
Nested variables in rules

### ISSUE THREAD

**Issue by petrusek** [goldibex/targaryen#24]
There is some undocumented problem with nested variables in Firebase rules recognized by targaryen. Firebase itself is handling them correctly.

Please see PR #23 where I demonstrate the problem in a (currently disabled) test suite.

```
Error: Expected an unauthenticated user to be able to read first/second/third, but the rules denied the read.
    /:<no rules>
    /first:<no rules>
    /first/second:<no rules>
    /first/second/third/: "$one == 'first' && $two == 'second' && $three == 'third'"
    $one: unknown variable $one
        => false
No .read rule allowed the operation.
Read was denied.
```

versus Firebase simulator

```
Attempt to read /first/second/third with auth=null
	/: "false"
		=> false
	/first
	/first/second
	/first/second/third: "$one == 'first' && $two == 'second' && $three == 'third'"
		=> true

Read was allowed.
```

**Comment by CurtisHumphrey** [comment]
To be able to repeat this we would also need a copy of your rules.json file too

**Comment by petrusek** [comment]
No problem, see the PR #23... it should be all there.

**Comment by bijoutrouvaille** [comment]
I did a little searching, and I think it has to do with this part https://github.com/goldibex/targaryen/blob/90a143318c6ad3770d6cfa85e267ef5e427a46db/lib/ruleset.js#L235

tryRead isn't propagating the wildchild'ren if the rule is null.

I've modified the part to be like this

    if (ruleDef.rule === null) {
      result.info += ':<no rules>\n';
      if (ruleDef.wildchild) {
        state[ruleDef.wildchild.name] = ruleDef.wildchild.value;
      }
    } else {

so now reads now work and the tests seem to pass.

**Comment by goldibex** [comment]
Fixed by #23.
